python/python/res/simulator/robust_simulator_context.py

```diff
@@ -0,0 +1,73 @@
+import time
+from collections import namedtuple
+from res.server import SimulationContext
+from res.enkf import NodeId, EnkfNode
+
+Status = namedtuple("Status", "running waiting complete failed")
+
+class RobustContext(SimulationContext):
+
+
+    def __init__(self, result_keys, ert, fs, mask, itr):
+        """
+        Handle which can be used to query status and results for robust simulation.
+        """
+        super(RobustContext, self).__init__(ert, fs, mask, itr)
+        self.result_keys = result_keys
+        self.res_config = ert.resConfig( )
+
+
+    def join(self):
+        """
+        Will block until the simulation is complete.
+        """
+        while self.isRunning():
+            time.sleep(1)
+
+
+    @property
+    def status(self):
+        """
+        Will return the state of the simulations.
+        """
+        return Status(running = self.getNumRunning(),
+                      waiting = self.getNumWaiting() + self.getNumPending(),
+                      complete = self.getNumSuccess(),
+                      failed = self.getNumFailed())
```

Comment:
why not reporting waiting and pending separately?

Reply:
Waiting; implies waiting in the internal `libres` queue system, whereas Pending means that it is waiting in the LSF queue system, Could have been reported separately, but seen from the user it it is much the same thing. 

Maybe it would have been better to report them separately at this stage; then everest could later choose to collapse them - or not.

Reply:
I have changed this - reporting both seperately.